### DISCUSSION TITLE
When using a custom title bar, if the mouse is clicked on the AreaTitleBar area to close it while the ComboBox are open, the entire application will lose focus.

### DISCUSSION THREAD

**Discussion by SmRiley** [AvaloniaUI/Avalonia#13668] [Q&A] [answered]
As shown in the video, the program will lose focus:

https://github.com/AvaloniaUI/Avalonia/assets/45205313/ac64151f-a4ff-435b-914e-cfc548d548ad

This is the code
```xaml
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:vm="using:AvaloniaApplication2.ViewModels"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaApplication2.Views.MainWindow"
        x:DataType="vm:MainWindowViewModel"
        Icon="/Assets/avalonia-logo.ico"
        Title="AvaloniaApplication2"
        ExtendClientAreaToDecorationsHint="True"
        ExtendClientAreaChromeHints="NoChrome"
        ExtendClientAreaTitleBarHeightHint="100">
	<StackPanel>
		<Button Content="test" Width="200" Height="200" Background="Red"/>
		<ComboBox>
			<ComboBoxItem>123</ComboBoxItem>
			<ComboBoxItem>456</ComboBoxItem>
			<ComboBoxItem>789</ComboBoxItem>
		</ComboBox>
	</StackPanel>
</Window>
```

**Comment by aniruddha76** [comment]
This behavior might be related to how Avalonia handles input events and focus.
One possible solution is to handle the mouse click event on the custom title bar and manually close the ComboBox before the application loses focus. You can achieve this by subscribing to the PointerPressed event on the title bar area and handling it in code-behind.

```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:vm="using:AvaloniaApplication2.ViewModels"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaApplication2.Views.MainWindow"
        x:DataType="vm:MainWindowViewModel"
        Icon="/Assets/avalonia-logo.ico"
        Title="AvaloniaApplication2"
        ExtendClientAreaToDecorationsHint="True"
        ExtendClientAreaChromeHints="NoChrome"
        ExtendClientAreaTitleBarHeightHint="100"
        PointerPressed="OnTitleBarPointerPressed">
    <StackPanel>
        <Button Content="test" Width="200" Height="200" Background="Red"/>
        <ComboBox x:Name="myComboBox">
            <ComboBoxItem>123</ComboBoxItem>
            <ComboBoxItem>456</ComboBoxItem>
            <ComboBoxItem>789</ComboBoxItem>
        </ComboBox>
    </StackPanel>
</Window>
```

hope this helps, then mark this as answer

**Reply by SmRiley** [reply]
how to close the ComboBox ? I use that code: `ComboBox.IsDropDownOpen = false;` but is useless

**Reply by timunie** [reply]
It does work for me. Maybe you need to make use of `Dispatcher.UiThread.Post`?

**Reply by SmRiley** [reply]
Can you see what's wrong with this?

![image](https://github.com/AvaloniaUI/Avalonia/assets/45205313/d1dc69bf-61b6-4921-a7c9-8d7ae1c3eece)

**Reply by timunie** [reply]
nothing obvious. does the event handler get actually hit if you set a break point? Probably one needs to tunnel the event? 🤔

**Reply by SmRiley** [reply]
yes it will, what do you mean by tunneling

**Reply by timunie** [reply]
https://docs.avaloniaui.net/docs/next/concepts/input/routed-events#routing-strategies

**Reply by SmRiley** [reply]
If added like this, it still not work. `this.AddHandler(Window.PointerPressedEvent, Window_PointerPressed, RoutingStrategies.Tunnel);`

**Reply by timunie** [reply]
Please file a minimal sample. I cannot reproduce on my side.

**Reply by SmRiley** [reply]
[Assets.zip](https://github.com/AvaloniaUI/Avalonia/files/13454739/Assets.zip)

**Reply by timunie** [reply, marked as answer]
Seems to be fixed in latest nightly. 

Note: Win32 impl was fixed, So behavior may change for you.
